Engine/Code/Engine/Graphics/Renderer.cpp

```diff
@@ -249,12 +251,27 @@ void Renderer::Drawtext(const Vec2& position, const Vec4& color, const std::stri
 
 		uvPos = font->GetGlyphUV(asciiText[i]);
 
-		mb->Push(VertexMaster(Vec3(quadPos.m_Mins.m_X, quadPos.m_Mins.m_Y, 0.0f), Vec4(color.m_X, color.m_Y, color.m_Z, color.m_W), Vec2(uvPos.m_Mins.m_X, uvPos.m_Maxs.m_Y)));
-		mb->Push(VertexMaster(Vec3(quadPos.m_Maxs.m_X, quadPos.m_Mins.m_Y, 0.0f), Vec4(color.m_X, color.m_Y, color.m_Z, color.m_W), Vec2(uvPos.m_Maxs.m_X, uvPos.m_Maxs.m_Y)));
-		mb->Push(VertexMaster(Vec3(quadPos.m_Maxs.m_X, quadPos.m_Maxs.m_Y, 0.0f), Vec4(color.m_X, color.m_Y, color.m_Z, color.m_W), Vec2(uvPos.m_Maxs.m_X, uvPos.m_Mins.m_Y)));
-		mb->Push(VertexMaster(Vec3(quadPos.m_Mins.m_X, quadPos.m_Maxs.m_Y, 0.0f), Vec4(color.m_X, color.m_Y, color.m_Z, color.m_W), Vec2(uvPos.m_Mins.m_X, uvPos.m_Mins.m_Y)));
-		
-		Mesh* mesh = mb->CreateMesh<VertexMaster::PCU>();
+		mb->Color3f(Vec4(color.m_X, color.m_Y, color.m_Z, color.m_W));
+	    mb->TexCoord2f(Vec2(uvPos.m_Mins.m_X, uvPos.m_Maxs.m_Y));
+	    mb->Position3f(Vec3(quadPos.m_Mins.m_X, quadPos.m_Mins.m_Y, 0.0f));
+	    mb->Push();
+	    
+	    mb->Color3f(Vec4(color.m_X, color.m_Y, color.m_Z, color.m_W));
+	    mb->TexCoord2f(Vec2(uvPos.m_Maxs.m_X, uvPos.m_Maxs.m_Y));
+	    mb->Position3f(Vec3(quadPos.m_Maxs.m_X, quadPos.m_Mins.m_Y, 0.0f));
+	    mb->Push();
+	    
+	    mb->Color3f(Vec4(color.m_X, color.m_Y, color.m_Z, color.m_W));
+	    mb->TexCoord2f(Vec2(uvPos.m_Maxs.m_X, uvPos.m_Mins.m_Y));
+	    mb->Position3f(Vec3(quadPos.m_Maxs.m_X, quadPos.m_Maxs.m_Y, 0.0f));
+	    mb->Push();
+	    
+	    mb->Color3f(Vec4(color.m_X, color.m_Y, color.m_Z, color.m_W));
+	    mb->TexCoord2f(Vec2(uvPos.m_Mins.m_X, uvPos.m_Mins.m_Y));
+	    mb->Position3f(Vec3(quadPos.m_Mins.m_X, quadPos.m_Maxs.m_Y, 0.0f));
+	    mb->Push();
```

Comment:
Seems good to me .. But if we're going by the way GL interfaces work, Adding position auto pushes the vertex to the array. glVertex3f usually functions that way, but keeping it this way is fine with me too. :)